concent_api/middleman/middleman_server.py

```diff
@@ -230,3 +237,32 @@ def _cancel_pending_tasks(self, tasks: Iterable[asyncio.Task], await_cancellatio
                 # Cancelled task raises asyncio.CancelledError that we can suppress:
                 with suppress(asyncio.CancelledError):
                     self._loop.run_until_complete(task)
+
+    async def _authenticate_signing_servcie(self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        logger.info("Signing Service candidate has connected, authenticating...")
+        authentication_task = self._loop.create_task(is_authenticated(reader, writer))
+        index = len(self._ss_connection_candidates)
```

Comment:
If this is used only once, maybe it is better to use it directly.

Reply:
I need to get the index of the last element of the list, then I modify it.